src/lightning_app/api/http_methods.py

```diff
@@ -19,6 +21,77 @@ def _signature_proxy_function():
     pass
 
 
+@dataclass
+class _FastApiMockRequest:
+    """This class is meant to mock FastAPI Request class that isn't pickle-able.
+
+    If a user relies on FastAPI Request annotation, the Lightning framework
+    patches the annotation before pickling and replace them right after.
+
+    Finally, the FastAPI request is converted back to the _FastApiMockRequest
+    before being delivered to the users.
+
+    Example:
+
+        import lightning as L
+        from fastapi import Request
+        from lightning.app.api import Post
+
+        class Flow(L.LightningFlow):
+
+            def request(self, request: Request) -> OutputRequestModel:
+                ...
+
+            def configure_api(self):
+                return [Post("/api/v1/request", self.request)]
+    """
+
+    _body: Optional[str] = None
+    _json: Optional[str] = None
+    _method: Optional[str] = None
+    _headers: Optional[Dict] = None
+
+    @property
+    def receive(self):
+        raise NotImplementedError
+
+    @property
+    def method(self):
+        raise self._method
+
+    @property
+    def headers(self):
+        return self._headers
+
+    def body(self):
```

Comment:
why the others are not property too?

Reply:
Because you can't easily support streaming across queues.